sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -118,6 +118,47 @@ class AstBuilder extends DataTypeAstBuilder
     }
   }
 
+  /**
+   * Override the base getIdentifierParts to properly parse qualified identifiers in
+   * IDENTIFIER('literal') contexts. Uses CatalystSqlParser to handle qualified identifiers
+   * like IDENTIFIER('`catalog`.`schema`') which should be parsed as Seq("catalog", "schema").
+   */
+  override protected def getIdentifierParts(ctx: ParserRuleContext): Seq[String] = {
+    ctx match {
+      case idCtx: IdentifierContext =>
```

Comment:
This seems duplicated with the similar looking code in DataTypeAstBuilder.scala, can we dedup it?